### PR TITLE
highlight users under 18 (minors)

### PR DESCRIPTION
**Pull request by Catalin12**

### CODE REVIEW

File: component-interaction/src/app/age-user-pipe.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import { Pipe, PipeTransform } from "@angular/core";
+
+@Pipe({ name: "ageUsers" })
```
Comment (danielursu) [inline component-interaction/src/app/age-user-pipe.ts]:
```suggestion
@Pipe({ name: "ageUser" })
```

File: component-interaction/src/app/age-user-pipe.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+export class AgeUserPipe implements PipeTransform {
+
+	public transform(date: any, ...args: any[]): number {
```
Comment (danielursu) [inline component-interaction/src/app/age-user-pipe.ts]:
```suggestion
	public transform(dateOfBirth: Date, ...args: any[]): number {
```
new Date(). and date. was confusing for me

File: component-interaction/src/app/user-details/user-details.component.html [hunk 3]
```diff
@@ -1,7 +1,8 @@
-<pre>
+<pre [ngClass]="{'adult': (user?.dateOfBirth | ageUsers) < 18}">
```
Comment (danielursu) [inline component-interaction/src/app/user-details/user-details.component.html]: wrap up intro a pipe IsAdultPipe

Comment (danielursu) [reply component-interaction/src/app/user-details/user-details.component.html]: in order to exercise the Pipes topic

**Review by danielursu — CHANGES_REQUESTED** [review]

File: component-interaction/src/app/user-details/user-details.component.html (hunk 3 above)
Comment (Catalin12) [reply component-interaction/src/app/user-details/user-details.component.html]: **_done_**

File: component-interaction/src/app/pipes/is-adult-user.pipe.ts [hunk 4]
```diff
@@ -0,0 +1,8 @@
+import { Pipe, PipeTransform } from "@angular/core";
+
+@Pipe({ name: "isAdult" })
+export class IsAdultUserPipe implements PipeTransform {
+	public transform(userAge: number) {
```
Comment (danielursu) [inline component-interaction/src/app/pipes/is-adult-user.pipe.ts:5]:
```suggestion
	public transform(userAge: number): boolean {
```
return type

File: component-interaction/src/app/user-list/user-list.component.html [hunk 5]
```diff
@@ -1,5 +1,4 @@
 <button class="succes" (click)="handleAddUserClick()">Insert user!</button>
 <button *ngIf="selectedUserIds.length" (click)="handleDeleteSelectedUsersClick()">Delete users</button>
-
-<user-details *ngFor="let user of users | sortUsers: TaskStatus.ASC" [user]="user"
+<user-details *ngFor="let user of (users | sortUsers: TaskStatus.ASC)" [user]="user"
```
Comment (danielursu) [inline component-interaction/src/app/user-list/user-list.component.html:3]:
```suggestion
<user-details *ngFor="let user of (users | sortUsers: orderType.ASC)" [user]="user"
```
naming of the file isn't the same with the Enum

**Review by danielursu — CHANGES_REQUESTED** [review]

**Pull request merged**